python/pyspark/tests/test_rdd.py

```diff
@@ -605,7 +605,7 @@ def test_distinct(self):
 
     def test_external_group_by_key(self):
         self.sc._conf.set("spark.python.worker.memory", "1m")
-        N = 200001
+        N = 2000001
```

Comment:
Mind explaining why it has to be changed?

Reply:
Well, if the object that you'd serialised is too large to fit in memory (like here https://github.com/apache/spark/blob/05cf81e6de3d61ddb0af81cd179665693f23351f/python/pyspark/shuffle.py#L774), then the `result.data` will have type `shuffle.ExternalListOfList`. If the object is small, then the type will be just `list`.
When using protocol version 4 (since the test fails only for Python 3.4) we serialise more efficiently, so we need more data not to fit in memory.
Is that enough to justify the change? @HyukjinKwon

Reply:
Yup, sounds good.